src/jit/core_templates.expr

```diff
@@ -35,6 +35,51 @@
 
 (template: getlex (copy $1))
 (template: bindlex! (store $0 $1 reg_sz))
+
+(template: getlex_ni
+  (load
+    (call (^func &MVM_frame_find_lexical_by_name)
+      (arglist
+        (carg (tc) ptr)
+        (carg (^cu_string $1) ptr)
+        (carg (const (&QUOTE MVM_reg_int64) int_sz) int)) int_sz)
+    int_sz))
+
+(template: getlex_nn
+  (load
+    (call (^func &MVM_frame_find_lexical_by_name)
+      (arglist
+        (carg (tc) ptr)
+        (carg (^cu_string $1) ptr)
+        (carg (const (&QUOTE MVM_reg_num64) int_sz) int)) int_sz)
+    int_sz))
+
+(template: getlex_ns
+  (load
+    (call (^func &MVM_frame_find_lexical_by_name)
+      (arglist
+        (carg (tc) ptr)
+        (carg (^cu_string $1) ptr)
+        (carg (const (&QUOTE MVM_reg_str) int_sz) int)) int_sz)
+    ptr_sz))
+
+(template: getlex_no
+  (let: (($found (call (^func &MVM_frame_find_lexical_by_name)
+                   (arglist
+                     (carg (tc) ptr)
+                     (carg (^cu_string $1) ptr)
+                     (carg (const (&QUOTE MVM_reg_obj) int_sz) int)) ptr_sz)))
+    (if (nz $found)
+      (do (when (nz (call (^func &MVM_spesh_log_is_logging)
+                      (arglist
+                        (carg (tc) ptr)) int_sz))
```

Comment:
I see now that I should not have called that int_sz (better int64_sz). MVM_spesh_is_logging nominally returns a 32-bit int, and you're telling the JIT that it is 64 bit, which will affect the `test` operation. That will not matter in this case - anytime we assign to the `rax` register as a 32 bit operand, we clear the upper operands anyway. But it is something to be aware of.

Reply:
Also, JIT-tted code should not be logging at all, so this while check can probably be removed.